3900 - splitting out legacy config page (#15093)
DataDog/documentation: content/en/infrastructure/livecontainers/legacy.md

---
title: Live Containers Legacy Configuration
kind: documentation
further_reading:
- link: "/infrastructure/livecontainers/configuration"
  tag: "Documentation"
  text: "Configure live containers"
---

## Overview

> This page provides instructions for setting up Live Containers for older Agent versions. These instructions apply to Datadog Agent versions [7.21.1][1] through 7.27.0 and Cluster Agent [1.9.0][2] through 1.11.0.

Reviewer: Are these links expected to open tagging docs?

> This page provides instructions for setting up Live Containers for older Agent versions. These instructions apply to Datadog Agent versions [7.21.1][1] through 7.27.0 and Cluster Agent [1.9.0][2] through 1.11.0.

Reviewer: It looks like these links currently lead to tagging documentation, is that expected?

{{< tabs >}}
{{% tab "Helm" %}}

If you are using the official [Datadog Helm Chart][1]:

- Use chart version above 2.4.5 and before 2.10.0. Starting from chart version 2.10.0 onwards, see the [latest configuration instructions][2] instead.
  **Note**: Ensure the Agent and Cluster Agent versions are hardcoded with the minimum versions required or above in your Helm chart [values.yaml][3] file.

> - Set `datadog.orchestratorExplorer.enabled` to `true` in [values.yaml][3]

Reviewer: ```suggestion
- Set `datadog.orchestratorExplorer.enabled` to `true` in [values.yaml][3].
```

> - Set `datadog.orchestratorExplorer.enabled` to `true` in [values.yaml][3]

Reviewer: ```suggestion
- Set `datadog.orchestratorExplorer.enabled` to `true` in [values.yaml][3].
```

- Deploy a new release.

In some setups, the Process Agent and Cluster Agent cannot automatically detect a Kubernetes cluster name. If this happens, the feature does not start, and the following warning displays in the Cluster Agent log: `Orchestrator explorer enabled but no cluster name set: disabling`. In this case, set `datadog.clusterName` to your cluster name in [values.yaml][3].

[1]: https://github.com/DataDog/helm-charts
[2]: /infrastructure/livecontainers/#configuration
[3]: https://github.com/DataDog/helm-charts/blob/master/charts/datadog/values.yaml
{{% /tab %}}
{{% tab "DaemonSet" %}}

The Cluster Agent must be running, and the Agent must be able to communicate with it. See the [Cluster Agent Setup][1] for configuration.

1. Set the Cluster Agent container with the following environment variable:

    ```yaml
      - name: DD_ORCHESTRATOR_EXPLORER_ENABLED
        value: "true"
    ```

2. Set the Cluster Agent ClusterRole with the following RBAC permissions.

    **Note**: For the `apps` apiGroups, Live Containers need permissions

>     to collect common kubernetes resources (`pods`, `services`, `nodes`, etc.),
>     which should be already in the RBAC if you followed [Cluster Agent Setup][1]. But if they are missing, ensure they are added (after `deployments`, `replicasets`):

Reviewer: ```suggestion
    to collect common Kubernetes resources (for example, `pods`, `services`, and `nodes`),
    which are already in the RBAC if you followed [Cluster Agent Setup][1]. If they are missing, ensure they are added (after `deployments`, `replicasets`):
```

>     to collect common kubernetes resources (`pods`, `services`, `nodes`, etc.),
>     which should be already in the RBAC if you followed [Cluster Agent Setup][1]. But if they are missing, ensure they are added (after `deployments`, `replicasets`):

Reviewer: ```suggestion
    to collect common Kubernetes resources (for example, `pods`, `services`, and `nodes`),
    which are already in the RBAC if you followed [Cluster Agent Setup][1]. If they are missing, ensure they are added (after `deployments`, `replicasets`):
```
